### PR TITLE
Uses rector to start modernising codebase

### PR DESCRIPTION
**Pull request by bensquire**
This PR uses `rector` to start modernising the codebase; currently focussing on adding mentod return types.

In addition I've added the missing composer extension requirements.

### CODE REVIEW

File: _lib/core/common.php [hunk 1]
```diff
@@ -543,7 +544,7 @@ function send_mail($opts = [])
 {
     require 'vendor/autoload.php'; // If you're using Composer (recommended)
     
-    $email = new \SendGrid\Mail\Mail();
+    $email = new Mail();
```
Comment (adamjimenez) [inline _lib/core/common.php]: will this still work?

Comment (bensquire) [reply _lib/core/common.php]: It should be fine as we have the import at the top of the page.

Comment (bensquire) [reply _lib/core/common.php]: @adamjimenez Where does that class actually live?

Comment (bensquire) [reply _lib/core/common.php]: Should it be a composer dependancy?

**Pull request merged**